### ISSUE TITLE
Lint empty curly braces with spaces and/or line-breaks

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2668]
cf. https://style.tidyverse.org/syntax.html?q=empty%20curly%20braces#indenting

``` r
library(lintr)

lint(
  text = "while (waiting_for_something()) { }",
  linters = all_linters()
)
#> ℹ No lints found.

lint(
  text = "while (waiting_for_something()) {

}",
  linters = all_linters()
)
#> ℹ No lints found.
```

<sup>Created on 2024-10-02 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Instead of no lint, the `{lintr}` should recommend:

```r
while (waiting_for_something()) {}
```

**Comment by IndrajeetPatil** [comment]
Ideally, `{styler}` takes care of this https://github.com/r-lib/styler/issues/1232

**Comment by MichaelChirico** [comment]
One thing to be cognizant of & test for:

```r
while (waiting()) {
  # comment
}
```

Should not lint. Just checked that {styler} ignores this too as expected.
